code/game/objects/effects/effect_system/effect_shield.dm

```diff
@@ -0,0 +1,26 @@
+/obj/effect/shield
+	/var/old_heat_capacity
+	/var/turf/location
+	name = "shield"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "wave2"
+	layer = ABOVE_NORMAL_TURF_LAYER
+	flags_1 = PREVENT_CLICK_UNDER_1
+
+/obj/effect/shield/Initialize()
+	. = ..()
+	location = get_turf(src)	
+	old_heat_capacity=location.heat_capacity
+	location.heat_capacity = INFINITY
+
+/obj/effect/shield/Destroy()
+	location = get_turf(src)	
+	location.heat_capacity=max(old_heat_capacity, location.heat_capacity)
```

Comment:
Isn't heat_capacity always going to be INFINITY? Just set it to old_heat_capacity.

Reply:
Ah yes, this was supposed to be in case tiles somehow got fucked with under the shield so a reinforced floor didn’t suddenly end up with only 10k heat capacity, but it only works for that particular scenario.